src/wallet/wallet.cpp

```diff
@@ -958,8 +958,9 @@ bool CWallet::LoadToWallet(const CWalletTx& wtxIn)
  * Abandoned state should probably be more carefully tracked via different
  * posInBlock signals or by checking mempool presence when necessary.
  */
-bool CWallet::AddToWalletIfInvolvingMe(const CTransaction& tx, const CBlockIndex* pIndex, int posInBlock, bool fUpdate)
+bool CWallet::AddToWalletIfInvolvingMe(const CTransactionRef& ptx, const CBlockIndex* pIndex, int posInBlock, bool fUpdate)
```

Comment:
The CWalletTx constructor is:
```c++
CWalletTx(const CWallet* pwalletIn, CTransactionRef arg) : CMerkleTx(std::move(arg))
```

I was suggesting to add a constructor

```c++
CWalletTx(const CWallet* pwalletIn, CTransactionRef&& arg) : CMerkleTx(arg)
```

and then update L984 to

```c++
            CWalletTx wtx(this, std::move(ptx));
```

and update this line to

```c++
bool CWallet::AddToWalletIfInvolvingMe(const CTransactionRef ptx, const CBlockIndex* pIndex, int posInBlock, bool fUpdate)
```

There are two reasons:
  - slightly more clear intent (AddToWalletIfInvolvingMe does get ownership)
  - Eliminates an indirection